src/librustdoc/clean/mod.rs

```diff
@@ -308,10 +307,9 @@ impl Clean<ExternalCrate> for CrateNum {
                 for attr in attrs.lists("doc") {
                     if let Some(v) = attr.value_str() {
                         if attr.check_name("keyword") {
-                            keyword = Keyword::from_str(&v.as_str()).ok()
-                                                                    .map(|x| x.name().to_string());
-                            if keyword.is_some() {
-                                break
+                            if Ident::with_empty_ctxt(v).is_reserved() {
+                                keyword = Some(v.to_string());
```

Comment:
I don't think the new code is equivalent to the old code. AFAICT, the old code accepts any keyword in the `Keywords` list in `symbol.rs`. The new code will behave differently, rejecting:
- `dyn`, `async`, `await`, `try` (if not 2018 edition)
- `'-`, `'static`, `auto`, `catch`, `default`, `existential`, `union` (unconditional)

Perhaps this difference doesn't matter in practice, I'm not sure.

Reply:
This check attempts to restrict identifiers that can be documented in https://doc.rust-lang.org/nightly/std/#keywords to a certain subset.
I'm not sure it is even necessary.
I'd rather remove it than add a method reproducing the previous behavior.
cc @GuillaumeGomez

Reply:
The old behavior is restored.